.github/workflows/build.yml

```diff
@@ -37,42 +37,7 @@ jobs:
     #! -------------------------------------------------------------------------
     - name: test-core-certifier-programs
       run: |
-        echo " "
-        echo "******************************************************************"
-        echo "* Check that core certifier programs still compile and clear tests"
-        echo "* (Also builds shared libraries for use by Python bindings.)"
-        echo "******************************************************************"
-        echo " "
-        pushd src
-
-        #! ---------------------------------------------------------------------
-        #! Check that core certifier programs still compile and clear tests
-        #! ---------------------------------------------------------------------
-        make -f certifier.mak clean
-        make -f certifier_tests.mak clean
-
-        make -j2 -f certifier.mak
-
-        # We need to clean here, otherwise make certifier_tests.mak will run
-        # into some protobuf-related errors.
-        make -f certifier.mak clean
-
-        make -j2 -f certifier_tests.mak
-        ./certifier_tests.exe
-
-        # Rebuild both shared libraries as 'all' target does not build these.
-        # Rebuild special-target which will invoke -fPIC flag during compilation.
-        set -x
-        make -f certifier.mak --always-make -j2 sharedlib
-        make -f certifier.mak --always-make -j2 swigpytestssharedlib
-        make -f certifier_tests.mak --always-make -j2 sharedlib
-        set +x
-
-        popd
-
-        #! Sanity check that the shared library was built correctly and can be loaded
-        #! python3 -c "import libcertifier_framework; help(libcertifier_framework)"
-        #! python3 -c "import libcertifier_tests; help(libcertifier_tests)"
+        ./CI/scripts/test.sh test-core-certifier-programs
```

Comment:
Everything will look just like this diff:

- Chunks of code from `build.yml` yanked into a function of the same name in `test.sh`
- Replace the test chunk code with a call to `test.sh` naming the test-case function